extensions/jdbc/jdbc-postgresql/runtime/src/main/java/io/quarkus/jdbc/postgresql/runtime/graal/SQLXLMFeature.java

```diff
@@ -0,0 +1,56 @@
+package io.quarkus.jdbc.postgresql.runtime.graal;
+
+import java.lang.reflect.Method;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import org.graalvm.nativeimage.hosted.Feature;
+import org.graalvm.nativeimage.hosted.RuntimeReflection;
+
+import com.oracle.svm.core.annotate.AutomaticFeature;
+
+@AutomaticFeature
+public final class SQLXLMFeature implements Feature {
+
+    private final AtomicBoolean triggered = new AtomicBoolean(false);
+
+    /**
+     * To set this, add `-J-Dio.quarkus.jdbc.postgresql.graalvm.diagnostics=true` to the native-image parameters
+     */
+    private static final boolean log = Boolean.getBoolean("io.quarkus.jdbc.postgresql.graalvm.diagnostics");
+
+    @Override
+    public void beforeAnalysis(BeforeAnalysisAccess access) {
+        Class<?> pgSQLXMLClass = access.findClassByName("io.quarkus.jdbc.postgresql.runtime.graal.PgSQLXML");
+        try {
+            //Only if this method is ever invoked, then it's possible that field {@link PgSQLXML#domResult }
+            //is ever non-null, and only in that case the XML parsing facilities pulled in by the driver
+            //end up being necessary.
+            final Method triggerMethod = pgSQLXMLClass.getMethod("setResult", Class.class);
+            access.registerReachabilityHandler(this::identifiedXMLProcessingInDriver, triggerMethod);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private void identifiedXMLProcessingInDriver(DuringAnalysisAccess duringAnalysisAccess) {
+        final boolean needsEnablingYet = triggered.compareAndSet(false, true);
+        if (needsEnablingYet) {
+            if (log) {
+                System.out.println(
+                        "Quarkus's automatic feature for GraalVM native images: enabling support for XML processing in the PosgreSQL JDBC driver");
```

Comment:
If we were using actual loggers, I would add a log somewhere when we know we will not enable support, but at the `debug`/`trace` level. Not sure how easy that would be, though. Maybe there is an `afterAnalysis` hook?

Reply:
yes, there is an `afterAnalysis`. Interesting idea, but I'd reserve that for the unified approach (I mentioned in the other comment).
We could also have a single logging know for all such features, rather than having a different system property each.